src/AggregateFunctions/AggregateFunctionMaxIntersections.h

```diff
@@ -117,7 +117,16 @@ class AggregateFunctionIntersectionsMax final
         const auto & value = this->data(place).value;
         size_t size = value.size();
         writeVarUInt(size, buf);
-        buf.write(reinterpret_cast<const char *>(value.data()), size * sizeof(value[0]));
+
+        for (size_t i = 0; i < size; ++i)
+        {
+            /// In this version, pairs were serialized with padding.
+            /// We must ensure that padding bytes are zero-filled.
```

Comment:
Do we need the same fix for `AggregateFunctionGroupArrayMoving` and `AggregateFunctionGroupArray`?
https://github.com/ClickHouse/ClickHouse/blob/70d1adfe4b7e437a62c59ca44a51fbd186ec7584/src/AggregateFunctions/AggregateFunctionGroupArrayMoving.h#L135
https://github.com/ClickHouse/ClickHouse/blob/70d1adfe4b7e437a62c59ca44a51fbd186ec7584/src/AggregateFunctions/AggregateFunctionGroupArray.h#L265